examples/platforms/nrf528xx/src/radio.c

```diff
@@ -457,6 +480,17 @@ otError otPlatRadioTransmit(otInstance *aInstance, otRadioFrame *aFrame)
 
     aFrame->mPsdu[-1] = aFrame->mLength;
 
+#if OPENTHREAD_CONFIG_ENABLE_PLATFORM_FEM_CUSTOM_SOURCE
+    if (nrf_802154_state_get() == NRF_802154_STATE_SLEEP)
+    {
+        // Enable FEM before RADIO leaving SLEEP state.
+        nrf5RadioCustomFemEnable();
+    }
+
+    // Adjust the output power based on the given channel.
+    nrf5RadioCustomFemAdjustPower(aFrame->mChannel, true);
```

Comment:
I might misunderstand the reason for this function but Radio Driver implements FAL layer. Wouldn't be better to replace `nrf_802154_fal_tx_power_get` function instead in your build? In the current implementation, I think that race conditions are possible when switching between TX/TX-ACK.

https://github.com/openthread/openthread/blob/master/third_party/NordicSemiconductor/drivers/radio/fal/nrf_802154_fal.c#L41

Possibly we can make the WEAK implementation of default function.

Reply:
The gain of the FEM is not a fixed value. The actual FEM gain is different on different channel. In order to ensure that the antenna  output power meets the FCC requirements, we need to adjust the radio output power and FEM gain on each channel.The function `nrf_802154_fal_tx_power_get ` only gets the radio output power and it can’t update the FEM gain. The function `nrf5RadioCustomFemAdjustPower ` is used to set the radio output power and FEM gain. The FEM gain remains the same if we don’t change it. The radio output power is maintained by the radio driver library after we set it. Could you provide more detailed info about the race conditions? I pinged a SED from a router with 1000 bytes length for 15000 times, I didn’t see the radio driver crash.

Reply:
Okay. When it comes to race conditions, I was more referring to the situation when we set output power for TX on channel X, but in the meantime, Radio Driver automatically replies with an ACK on channel Y with configured power for channel X. Is this possible?

Reply:
Thanks for your detailed info about the race condition. I understand your concern now. The race condition does exist. Your suggestion about using `nrf_802154_fal_tx_power_get` is great. After I verified your suggestion, I will update the code.

Reply:
I have moved all FEM APIs to the header file `platform-fem.h`, so that we don't need to add an extra macro to control the FEM API implementation. User can just implement all the FEM APIs in their own `fem.c`.  I also set the function `nrf_802154_fal_tx_power_get` to a weak function according to your suggestion.